### PR TITLE
(#1898) Adds External Link Icon to Feature Cards

### LINKED ISSUE #1898: Story: Add External Link Icon to NCIDS Feature Cards
### Specifications

**Visual**

- External Link icon will be center aligned vertically on the title text
- If the title would wrap, the last word of the title will wrap with the icon so the icon does not sit alone
- External link icons shall always appear on the right side of the title

**Functional**
- If the link text does not have `.gov` it will get an external link icon
- A Feature Card can be overridden to have an external link icon
- A Feature Card with an external link icon can be overridden to remove it

**Impacted Components**

- [ ] NCIDS Feature Card
  - [ ] Documentation Site update [Card Page](https://designsystem-dev.cancer.gov/develop/components/card)
  - [ ] [Storybook for Card](https://designsystem-dev.cancer.gov/develop/storybook/?path=/story/components-card-default--default) The new story will be added as a test case.

### DoR Tasks

- [x] Comps Created
- [x] Comps Reviewed
  - [x] Developer Review
  - [x] IA Review
  - [x] Product Review
- [x] Comps Approved
  - [x] Developer Approval
  - [x] IA Approval
  - [x] Product Approval
- [x] Scenarios Created
- [x] Scenarios Reviewed
  - [x] Designer Review
  - [x] Developer Review
  - [x] IA Review
- [x] Scenarios Approved
  - [x] Designer Approval
  - [x] Developer Approval
  - [x] IA Approval
- [x] Acceptance Criteria Drafted
- [x] Acceptance Criteria Reviewed
  - [x] Developer Review
  - [x] IA Review
- [x] Acceptance Criteria Approved
  - [x] Developer Approval
  - [x] IA Approval

### DoD Tasks

- [ ] Governance Drafted
- [ ] PR Review (see PR for review and comments)
- [ ] Client review with demo

### Technical Notes
**Scenarios/Comps**
- [Comp to base the designs on the Doc Site and Storybook on](https://www.figma.com/design/GCOniv8OTyj9OC1udKwo6S/Mini-Landing-Pages?node-id=8420-41386&node-type=text&m=dev)
- On the [Documentation Site Card page](https://designsystem-dev.cancer.gov/develop/components/card) add:
  - Add a subsection to [Best Practices](https://designsystem-dev.cancer.gov/develop/components/card#patterns) which will be the last subsection within Best Practices with the heading, "External Links", and in the subsequent paragraph explain how and when to use and add the External Link Icon. In this section, explain how the external links are mostly "auto-magical", and provide an explanation of how to use the override.
  - A code snippet after the [Card](https://designsystem-dev.cancer.gov/develop/components/card#card) showing Card with External Link Icon with an example and the code to create it, with the "no image" filler image, a link to www.google.com, a title of "External Link Example", and a description of, "Click on this card to go to Google."
- On Storybook:
  - Within the [Card folder](https://designsystem-dev.cancer.gov/develop/storybook/?path=/story/components-card-default--default), in the Test Cases folder, add a story called "Cards With External Links", and in the story there is a card row. The cards in the row will be:
  - Card 1
    -  Image: added to this ticket by the designers
    - Title: "External Link Example"
    - Description: "Click on this card to go to Google"
    - Link: www.google.com
  - Card 2
    - Image: added to this ticket by the designers
    - Title: "Override Internal to Show External"
    - Description: "This feature card has an internal link, but is using the override class to display the external link icon."
    - Link: www.cancer.gov
  - Card 3
    - Image: added to this ticket by the designers
    - Title: "Override External to Not Show Icon"
    - Description: "This feature card has an external link, but is using the override class to not display the external link icon."
    - Link: www.google.com

### Acceptance Criteria

```gherkin
Scenario: A card with an external link is viewed on the NCIDS
  Given the external link icon has been added to the NCIDS Documentation Site and NCIDS Storybook for cards
    And the card is linked to a url that does not contain `.gov`
  When a card has a linked url that does not contain `.gov`
  Then the card will appear with the external link icon
    And the icon on the card will match the comp (see link to comp above)
    And the external link icon will remain center aligned vertically on the line of text it is on
  When the text of the card title wraps
    And the width of the card is only narrow enough where the external link icon should wrap alone
  Then the last word of the title will wrap to the next line with the external link icon
  When a user wants to override the behavior of a card with a link that does not contain `.gov` to not display the external link icon
  Then they will be able to override the behavior
    And the external link icon will not be displayed
  When a user has a card with a link that contains `.gov`
    And they want to override the behavior to include the external link icon
  Then they will be able to override the behavior
    And the external link icon will be displayed
```

### PR DESCRIPTION
**Pull request by dlescarbeau**
## Pull Request Details

UPDATES SASS VERSION TO 1.69.0 AS A PEER DEPENDENCY

- Adds mixin for adding an external link icon to external links
- Utilizes new mixin to add external link icon to feature cards
- Updates `nci-card` twig file to fix an issue with its alt-text
- Updates `nci-card-group` twig file to accommodate the `modifier` that can be passed to `nci-card`

Closes #1898>

## Author PR Checklist
Items that the author of the PR is responsible for checking before submitted the PR.

### General:
- [x] I have reviewed the acceptance criteria defined in the ticket and ensured the work has been completed.
- [x] The commit message passes [all quality commit message standards](https://github.com/NCIOCPL/ncids/wiki/Creating-Quality-Commit-Messages).
- [x] Unit tests have been updated or created to reflect any javascript changes.
- [x] Storybook scenarios have been updated or created to reflect any html/css/js changes.

### Accessibility:
- [ ] WCAG 2.1 Level AA requirements have been met.

### Development:
- [x] Any new or updated javascript code has 100% unit test coverage.
- [x] New or updated breakpoints have regression images.
- [x] Breaking changes have been thoroughly documented in the PR.

## Product Reviewer PR Checklist
Items the product team is responsible for reviewing.

### General:
- [x] There are no unexpected or unapproved regression image changes. 
- [x] Functionality of interactive elements meet the acceptance criteria.
- [x] The product is visually and functionally the same across the different browsers.

### Accessibility:
- [x] AxeDev Tools: there are no new or outstanding accessibility issues introduced in this PR.
- [x] Lighthouse: scores have not noticeably decreased during this PR.
- [x] Wave: there are no new errors or contrast errors introduced in this PR.

## Design Reviewer PR Checklist
Items the design team is responsible for reviewing.  

### General:
- [ ] New or updated features introduced in this PR are developed mobile-first.
- [ ] Breakpoint changes and regression images match those breakpoints.
- [ ] This PR has been tested in all supported browsers at all breakpoints.

## Developer Reviewer PR Checklist
Items the development team is responsible for reviewing.

### General:
- [ ] New code passes code quality standards set by industry standards.
- [ ] The expected Storybook stories have been added or updated for the new or updated feature.
- [ ] The expected unit tests have been added or updated for the new or updated feature.

### Accessibility:
- [ ] VoiceOver: Described content matches with what was expected.
- [ ] Keyboard navigation: new or updated features and content are navigable via the keyboard.

### CODE REVIEW

**Comment by andyvanavery31** [comment]
Notes/Changes Requested:
- All three of the External Link Icon examples will be shown on the Documentation Site (with External Link, with External Link Override to Not Show Icon, and Internal Link Overridden to Show Icon)
- Rename the Code Snippets to "Card with External Link", "Card with Internal Link with External Link Override", and "Card with External Link with Icon Removed Override"
- Update the code snippets to use the content from the Scenarios section of the ticket to reduce user confusion.
- Update storybook story from "External Cards" to "Cards with External Links"

**Comment by dlescarbeau** [comment]
> Notes/Changes Requested:
> 
> * All three of the External Link Icon examples will be shown on the Documentation Site (with External Link, with External Link Override to Not Show Icon, and Internal Link Overridden to Show Icon)
> * Rename the Code Snippets to "Card with External Link", "Card with Internal Link with External Link Override", and "Card with External Link with Icon Removed Override"
> * Update the code snippets to use the content from the Scenarios section of the ticket to reduce user confusion.
> * Update storybook story from "External Cards" to "Cards with External Links"

- updated titles of code snippets per above to be "Card with External Link", "Card with Internal Link with External Icon Override", and "Card with External Link with Icon Removed Override"
- updated code snippet content
- updated story in storybook to be "Cards with External Links"

File: packages/ncids-css/packages/nci-card/internal/_nci-card.scss [hunk 1]
```diff
@@ -1,9 +1,33 @@
 @use 'uswds-core' as *;
+@use "sass:meta" as meta;
 @use '../../../internal-uswds/usa-card/src/styles/usa-card' as card;
 
+// Mixin to pass for overriding certain external icon styles
+@mixin external-icon() {
```
Comment (bryanpizzillo) [inline packages/ncids-css/packages/nci-card/internal/_nci-card.scss:6]: can you make these mixins something like "nci-card-external-icon" with the component name? Just so we don't get confused...

Comment (dlescarbeau) [reply packages/ncids-css/packages/nci-card/internal/_nci-card.scss:6]: I thought it being a mixin only used inside `_nci-card.scss` made that pretty obvious, but sure, whatever's best.

File: patches/@uswds__uswds@3.7.1.patch [hunk 2]
```diff
@@ -1278,6 +1278,165 @@ index 983d2ed46d758291a903957e79df24c9043260dd..40a1f924922b3dd40129c28e453e51ea
...
+-
+-$icon-object: (
+-  "name": "launch",
+-  "color": currentColor,
+-  "height": $external-link-size,
+-  "svg-height": 24,
+-  "svg-width": 24,
+-  "position-x": center,
+-  "position-y": center,
+-);
```
[the remark this answers is not in the document]
Comment (dlescarbeau) [reply patches/@uswds__uswds@3.7.1.patch:1301]: sure

**Comment by adrianacastaneda** [comment]
IA approval from both me and Lindsay! Thank you!

**Review by andyvanavery31 — APPROVED** [review]

**Comment by blilianyu** [comment]
Looks great thanks!

File: packages/ncids-css/package.json [hunk 3]
```diff
@@ -42,7 +42,7 @@
 		"postcss": "~8.4.33",
 		"postcss-loader": "6.1.1",
 		"rimraf": "~3.0.2",
-		"sass": "~1.49.0",
+		"sass": "~1.69.0",
```
Comment (bryanpizzillo) [inline packages/ncids-css/package.json:45]: We need to also set a peerDependency since you actually need to have 1.69 for this to work.

**Review by bryanpizzillo — APPROVED** [review]
Looks good!

**Review by andyvanavery31 — APPROVED** [review]

**Pull request merged**